### PR TITLE
update(userspace/libsinsp): add ppm_sc_code search in filter ASTs 

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area libsinsp

/area tests

**Does this PR require a change in the driver versions?**

This adds a new filter AST operation to search the set of ppm_sc_code matching a certain AST, just like we already had for ppm_event_code. This is required because relying entirely on ppm_event_codes might cause information loss on generic events (e.g. when finding PPME_GENERIC_E/X, we have no way to know _which_ generic event that really was).

Unit tests related to that code area have been extended to increase the coverage and avoid potential corner cases.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
update(userspace/libsinsp): add ppm_sc_code search in filter ASTs 
```

### CODE REVIEW

File: userspace/libsinsp/test/filter_ppm_codes.ut.cpp
Comment (jasondellaluce) [inline userspace/libsinsp/test/filter_ppm_codes.ut.cpp:96]: These two tests must be enabled once we merge https://github.com/falcosecurity/libs/pull/889.
Until then, the conversion will have information loss for old event versions, however I didn't feel like blocking this PR due to these failing tests, as the effort is not correlated to the one of #889.

**Review by Andreagit97 — APPROVED** [review]
/approve
Amazing :heart_eyes:

**Review by incertum — APPROVED** [review]
/approve

Amazing work as always :star_struck: let's merge and iterate should we find out something is still missing.

**Pull request merged**
